oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/StorageID.java

```diff
@@ -0,0 +1,157 @@
+        return this;
+    }
+
+    public StorageID append(String name, long value) {
+        if (StringUtil.isBlank(name)) {
+            throw new IllegalArgumentException("The name of storage ID should not be null or empty.");
+        }
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, name=" + name);
+        }
+        fragments.add(new Fragment(name, Long.class, false, value));
+        return this;
+    }
+
+    public StorageID append(String name, int value) {
+        if (StringUtil.isBlank(name)) {
+            throw new IllegalArgumentException("The name of storage ID should not be null or empty.");
+        }
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, name=" + name);
+        }
+        fragments.add(new Fragment(name, Integer.class, false, value));
+        return this;
+    }
+
+    public StorageID appendMutant(String source, long value) {
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, source=" + source);
+        }
+        fragments.add(new Fragment(source, Long.class, true, value));
+        return this;
+    }
+
+    public StorageID appendMutant(final String source, final String value) {
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, source=" + source);
+        }
+        fragments.add(new Fragment(source, String.class, true, value));
+        return this;
+    }
+
+    /**
+     * @return the string ID concatenating the values of {@link #fragments} by the underline(_).
+     */
+    public String build() {
+        sealed = true;
+        return Joiner.on(Const.ID_CONNECTOR).join(fragments);
+    }
+
+    /**
+     * @return a read-only list to avoid unexpected change for metric ID.
+     */
+    public List<Fragment> read() {
+        sealed = true;
+        return Collections.unmodifiableList(fragments);
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    @EqualsAndHashCode(of = {
+        "name",
+        "value"
+    })
+    public static class Fragment {
+        /**
+         * The column name of the value, or the original column name of the mutate value.
+         * This could be null if {@link #mutate} is true and no relative column, such as the original value is not in
+         * the persistence.
+         */
+        private final String name;
+        /**
+         * Represent the class type of the {@link #value}.
+         */
+        private final Class<?> type;
+        /**
+         * If true, the field was from {@link #name}, and value is changed by internal rules.
+         * Such as time bucket downsampling, use a day-level time-bucket to build the ID for a minute dimension metric.
+         */
+        private final boolean mutate;
+        private final Object value;
```

Comment:
@hanahmily @lujiajing1126 You would need this when you try to build BanyanDB series ID without string ID.

Notice, we should check whether there is conflict(s) between series IDs and fragments.

Reply:
It looks good to me.